### PR TITLE
chore: Bump Buf and Go versions

### PR DESCRIPTION
**Pull request by codingllama**
Update Buf and Go to the latest releases.

* https://github.com/bufbuild/buf/releases/tag/v1.21.0
* https://go.dev/doc/devel/release#go1.20.minor

Devbox:

* https://github.com/NixOS/nixpkgs/blob/master/pkgs/development/tools/buf/default.nix#L13
* https://github.com/NixOS/nixpkgs/blob/master/pkgs/development/compilers/go/1.20.nix#L49 (still on 1.20.4)

### CODE REVIEW

**Review by espadolini — APPROVED** [review]

File: .drone.yml [hunk 1]
```diff
@@ -39,7 +39,7 @@ name: push-build-linux-amd64
 environment:
   BUILDBOX_VERSION: teleport14
   GID: "1000"
-  RUNTIME: go1.20.4
+  RUNTIME: go1.20.5
```
Comment (jakule) [inline .drone.yml:42]: Are we not concerned by https://groups.google.com/g/golang-announce/c/q5135a9d924 ?

Comment (codingllama) [reply .drone.yml:42]: I don't think we use `gccgo` or GoLLVM. Unless I'm mistaken we should be fine.

Comment (jakule) [reply .drone.yml:42]: Go 1.20.5 "works for me", so we should be fine.

**Review by jakule — APPROVED** [review]

**Review by zmb3 — APPROVED** [review]

File: .drone.yml (hunk 1 above)
Comment (codingllama) [reply .drone.yml:42]: I haven't had problems with the libfido2 or touchid stuff, so assuming it's fine too. If have any problems, just rollback. ;)

**Comment by codingllama** [comment]
Thanks, folks.

**Pull request merged**